src/Agent.Worker/AgentPluginManager.cs

```diff
@@ -20,96 +18,42 @@ namespace Microsoft.VisualStudio.Services.Agent.Worker
         private readonly Dictionary<Guid, List<string>> _supportedTasks = new Dictionary<Guid, List<string>>();
 
-        private readonly HashSet<string> _taskPlugins = new HashSet<string>()
-        {
-            "Agent.Plugins.Repository.CheckoutTask, Agent.Plugins",
-            "Agent.Plugins.Repository.CleanupTask, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTask, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.PublishPipelineArtifactTask, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.PublishPipelineArtifactTaskV1, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTaskV1, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTaskV1_1_0, Agent.Plugins",
-            "Agent.Plugins.PipelineCache.SavePipelineCacheV0, Agent.Plugins",
-            "Agent.Plugins.PipelineCache.RestorePipelineCacheV0, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTaskV1_1_1, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTaskV1_1_2, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTaskV1_1_3, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTaskV2_0_0, Agent.Plugins",
-            "Agent.Plugins.PipelineArtifact.PublishPipelineArtifactTaskV0_140_0, Agent.Plugins"
-        };
-
-        private readonly HashSet<string> _commandPlugins = new HashSet<string>();
+        private HashSet<string> _taskPlugins = new HashSet<string>();
 
         public override void Initialize(IHostContext hostContext)
         {
             base.Initialize(hostContext);
-
-            // Load task plugins
-            foreach (var pluginTypeName in _taskPlugins)
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
             {
-                IAgentTaskPlugin taskPlugin = null;
-                AssemblyLoadContext.Default.Resolving += ResolveAssembly;
-                try
-                {
-                    Trace.Info($"Load task plugin from '{pluginTypeName}'.");
-                    Type type = Type.GetType(pluginTypeName, throwOnError: true);
-                    taskPlugin = Activator.CreateInstance(type) as IAgentTaskPlugin;
-                }
-                finally
-                {
-                    AssemblyLoadContext.Default.Resolving -= ResolveAssembly;
-                }
-
-                ArgUtil.NotNull(taskPlugin, nameof(taskPlugin));
-                ArgUtil.NotNull(taskPlugin.Id, nameof(taskPlugin.Id));
-                ArgUtil.NotNullOrEmpty(taskPlugin.Stage, nameof(taskPlugin.Stage));
-                if (!_supportedTasks.ContainsKey(taskPlugin.Id))
+                if (assembly.FullName.StartsWith("Agent.", StringComparison.OrdinalIgnoreCase))
```

Comment:
`Agent.Plugins.`?

Reply:
I am fine doing that but was concerned that we could move things around into other assemblies that we owned. I guess we would catch that in testing.